src/Cache.hs

```diff
@@ -0,0 +1,47 @@
+module Cache
+    ( cacheBoards
+    ) where
+
+import System.Environment (getEnv)
+
+import qualified Data.Text as T
+import qualified Data.ByteString.Lazy as BS
+import Control.Lens ((.~), (^.), (&))
+import LoadEnv (loadEnv)
+import Network.Wreq
+
+data Credentials = Credentials
+    { apiKey :: String
+    , apiToken :: String
+    , username :: String
+    }
+
+cacheBoards :: IO ()
+cacheBoards = do
+    loadEnv
+    c <- getCredentials
+    let opts' = opts c
+    let endpoint = boardsEndpoint $ username c
+    writeResponse opts' endpoint
+
+getCredentials :: IO Credentials
+getCredentials =
+    Credentials
+        <$> getEnv "TRELLO_API_KEY"
+        <*> getEnv "TRELLO_API_TOKEN"
+        <*> getEnv "TRELLO_USERNAME"
+
+opts :: Credentials -> Options
+opts (Credentials apiKey apiToken _) =
```

Comment:
It looks like this is the only thing that takes Credentials, and the final term (`TRELLO_USERNAME`) is unused. What if Credentials didn't have a `username` field at all?